File: hashTable/1207_unique_number_of_occurrences.py
class Solution:
    # O(n) time | O(n) space - where n is the length of the input array
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        freq = Counter(arr)
        seenFreq = set()
        for count in freq.values():
            if count in seenFreq:
                return False
            seenFreq.add(count)
        
        return True

File: hashTable/1207_unique_number_of_occurrences.go
// O(n) time | O(n) space - where n is the length of the input array
func uniqueOccurrences(arr []int) bool {
    freq := make(map[int]int)
    for _, val := range arr {
        freq[val]++
    }
    seenFreq := make(map[int]bool)
    for _, v := range freq {
        if _, exist := seenFreq[v]; exist {
            return false
        }
        seenFreq[v] = true
    }
    return true
}
